slasher/detection/attestations/spanner.go

```diff
@@ -203,58 +204,93 @@ func (s *SpanDetector) updateMinSpan(ctx context.Context, att *ethpb.IndexedAtte
 	spanMap := make(map[uint64]types.Span)
 	epochsSpansMap := make(map[uint64]map[uint64]types.Span)
 	epoch := source - 1
+	untilEpoch := epoch - epochLookback
+	if int(untilEpoch) < 0 || featureconfig.Get().DisableLookback {
+		untilEpoch = 0
+	}
 	useCache := true
 	useDb := false
 	var err error
-	for ; epoch >= 0; epoch-- {
-		if useCache {
-			spanMap, useCache, err = s.slasherDB.EpochSpansMap(ctx, epoch)
-		}
-		// Should happen once when cache is exhausted.
-		if !useCache && !useDb {
-			epochsSpansMap, err = s.slasherDB.EpochsSpanByValidatorsIndices(ctx, valIndices, epoch)
-			useDb = true
-		}
-		if err != nil {
-			return err
-		}
-		if useDb {
-			spanMap = epochsSpansMap[epoch]
-			if spanMap == nil {
-				spanMap = make(map[uint64]types.Span)
-				epochsSpansMap[epoch] = spanMap
+	for ; epoch >= untilEpoch; epoch-- {
+		if featureconfig.Get().DisableLookback {
```

Comment:
This seems like it when things go wrong it could run forever. Have you considered adding a context time out error here?

Reply:
that's a good idea, we can do if ctx.Err() != nil return pre-emptively

Reply:
Yep!